waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/exchange/ExchangeExternalSignaturesController.kt

```diff
@@ -0,0 +1,398 @@
+                authResponseURL,
+                parameters {
+                    formParams.forEach { entry ->
+                        entry.value.forEach { append(entry.key, it) }
+                    }
+                })
+
+            val responseBody = runCatching { resp.bodyAsText() }.getOrNull()
+            val isResponseRedirectUrl = responseBody != null && responseBody.take(10).lowercase().let {
+                @Suppress("HttpUrlsUsage")
+                it.startsWith("http://") || it.startsWith("https://")
+            }
+            logger.debug { "HTTP Response: $resp, body: $responseBody" }
+
+            val walletService = getWalletService()
+            val credentialService = CredentialsService()
+            presentedCredentialIdList.forEach {
+                credentialService.get(walletService.walletId, it)?.run {
+                    eventUseCase.log(
+                        action = EventType.Credential.Present,
+                        originator = authReq.clientMetadata?.clientName
+                            ?: EventDataNotAvailable,
+                        tenant = walletService.tenant,
+                        accountId = walletService.accountId,
+                        walletId = walletService.walletId,
+                        data = eventUseCase.credentialEventData(
+                            credential = this,
+                            subject = eventUseCase.subjectData(this),
+                            organization = eventUseCase.verifierData(authReq),
+                            type = null
+                        ),
+                        credentialId = this.id,
+                    )
+                }
+            }
+
+            val result = if (resp.status.value == 302 && !resp.headers["location"].toString().contains("error")) {
+                Result.success(if (isResponseRedirectUrl) responseBody else null)
+            } else if (resp.status.isSuccess()) {
+                Result.success(if (isResponseRedirectUrl) responseBody else null)
+            } else {
+                if (isResponseRedirectUrl) {
+                    Result.failure(
+                        PresentationError(
+                            message = "Presentation failed - redirecting to error page",
+                            redirectUri = responseBody
+                        )
+                    )
+                } else {
+                    logger.debug { "Response body: $responseBody" }
+                    Result.failure(
+                        PresentationError(
+                            message =
+                            if (responseBody != null) "Presentation failed:\n $responseBody"
+                            else "Presentation failed",
+                            redirectUri = ""
+                        )
+                    )
+                }
+            }
+
+            if (result.isSuccess) {
+//                wallet.addOperationHistory(
```

Comment:
perhaps cleanup and have a ticket for it to implement event logging for external-signature endpoints

Reply:
I already have this is a to do task in the respective notion track.